vortex-array/src/compute/take.rs

```diff
@@ -151,3 +213,31 @@ fn take_impl(array: &Array, indices: &Array, checked_indices: bool) -> VortexRes
         canonical_take_fn.take(&canonical, indices)
     }
 }
+
+fn take_into_impl(
+    array: &Array,
+    indices: &Array,
+    builder: &mut dyn ArrayBuilder,
+) -> VortexResult<()> {
+    if array.dtype() != builder.dtype() {
+        vortex_bail!(
+            "TakeIntoFn {} had a builder with a different dtype {} to the array dtype {}",
+            array.encoding(),
+            array.dtype(),
+            builder.dtype()
+        );
+    }
+    if let Some(take_fn) = array.vtable().take_fn() {
+        return take_fn.take_into(array, indices, builder);
+    }
+
+    // Otherwise, flatten and try again.
+    log::debug!("No take implementation found for {}", array.encoding());
```

Comment:
take_into?

Reply:
They share a trait so I guess it is TakeFn?